src/DearImGui.hs

```diff
@@ -570,21 +570,58 @@ arrowButton strId dir = liftIO do
   withCString strId \strIdPtr ->
     Raw.arrowButton strIdPtr dir
 
-
 -- | Wraps @ImGui::Checkbox()@.
 checkbox :: (HasSetter ref Bool, HasGetter ref Bool, MonadIO m) => String -> ref -> m Bool
-checkbox label ref = liftIO do
-  currentValue <- get ref
-  with (bool 0 1 currentValue) \boolPtr -> do
-    changed <- withCString label \labelPtr ->
-      Raw.checkbox labelPtr boolPtr
-
-    when changed do
-      newValue <- peek boolPtr
-      ref $=! (newValue == 1)
-
-    return changed
+checkbox label ref = stateful ref $ checkboxM label
```

Comment:
The wrapped functions aren't even needed if widgets are to have their "mutable" arguments stated last.

There can be `statefulN`-family for widgets that mutate N arguments at once.

```haskell
-- With all arguments expanded
dragFloatRange2 desc speed minValue maxValue minFmt maxFmt refMin refMax = ...

-- Just add your values
dragFloatRange01 label = dragFloatRange2 label 0.01 0 1 "%.1f" "%.1f"

action rMin rMax = do
  changed <- stateful2 rMin rMax $
    dragFloatRange01 "from zero to hero"
  when changed do
    ...
```